internal/events/types/charity/charity_event.go

```diff
@@ -0,0 +1,204 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package charity
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/twitchdev/twitch-cli/internal/events"
+	"github.com/twitchdev/twitch-cli/internal/models"
+	"github.com/twitchdev/twitch-cli/internal/util"
+)
+
+var transportsSupported = map[string]bool{
+	models.TransportEventSub: true,
+}
+var triggers = []string{"charity-donate", "charity-start", "charity-progress", "charity-stop"}
+
+var triggerMapping = map[string]map[string]string{
+	models.TransportEventSub: {
+		"charity-donate":   "channel.charity_campaign.donate",
+		"charity-start":    "channel.charity_campaign.start",
+		"charity-progress": "channel.charity_campaign.progress",
+		"charity-stop":     "channel.charity_campaign.stop",
+	},
+}
+
+type Event struct{}
+
+func (e Event) GenerateEvent(params events.MockEventParameters) (events.MockEventResponse, error) {
+	var event []byte
+	var err error
+	var campaign_id *string // only used by channel.charity_campaign.donate
+	var id *string          // used by the rest of channel.charity_campaign.*
+	var user_id *string
+	var user_login_name *string
+	var charity_description *string
+	var charity_website *string
+	var amount *models.CharityEventSubEventAmount
+	var current_amount *models.CharityEventSubEventAmount
+	var target_amount *models.CharityEventSubEventAmount
+	var started_at *string
+	var stopped_at *string
+
+	randomID := util.RandomGUID()
+	charityName := "Example Charity"
+	charityLogo := "https://abc.cloudfront.net/ppgf/1000/100.png"
+	charityDescription := "Example Description"
+	charityWebsite := "https://www.example.com"
+
+	if params.Trigger == "charity-donate" {
+		campaign_id = &randomID
+		user_id = &params.FromUserID
+		user_login_name = &params.FromUserName
+		amount = &models.CharityEventSubEventAmount{
+			Value:         10000,
+			DecimalPlaces: 2,
+			Currency:      "USD",
+		}
+	}
+	if params.Trigger == "charity-start" {
+		id = &randomID
+		charity_description = &charityDescription
+		charity_website = &charityWebsite
+		current_amount = &models.CharityEventSubEventAmount{
+			Value:         0,

```

Comment:
start might not always be 0 if yo uare resuming.

Might need a "restart" example payload.

Reply:
Fixed it by adding flags